### PR TITLE
Interactive list

### PR DESCRIPTION
**Pull request by Kaiserabbas**
Hi Dear Reviewers, Welcome to my pull request for TodoList(interactive lists)
- Added a new JavaScript file and import it as a module:
- Add event listener to the checkbox (change).
- Updated items object's value for completed key upon user actions.
- Implemented a function for the "Clear all completed" button.
- Stored the updated array of items in local storage, so the user gets the correct list of values after the page reloads).
Previous pull request
https://github.com/Kaiserabbas/todo-list/pull/1

### CODE REVIEW

File: src/modules/print.js [hunk 1]
```diff
@@ -115,54 +113,47 @@ export const showTasks = () => {
...
+    // Update the text decoration style based on the checkbox state
+    if (inputCheckBox.checked) {
+      listItemTitle.style.textDecoration = 'line-through';
+    } else {
+      listItemTitle.style.textDecoration = 'none';
+    }
+  }
   const clearCompletedButton = document.createElement('button');
   clearCompletedButton.setAttribute('class', 'clear-button');
   clearCompletedButton.textContent = 'Clear All Completed';
```
Comment (divinecharlotte) [inline src/modules/print.js:160]: - [ ] - [ ] Kindly consider implementing   Array.filter() on this ` clearCompletedButton ` to meet the below requirement
> Implement a function for the "Clear all completed" button (use filter() method).

File: src/modules/print.js [hunk 2]
```diff
@@ -115,54 +113,47 @@ export const showTasks = () => {
     listItemTitle.innerHTML = `${task.description}`;
     tasksDiv.appendChild(listItemTitle);
 
-    // Create a button to remove the book
     const removeButton = document.createElement('span');
     removeButton.setAttribute('class', 'close');
-    removeButton.getAttribute('id', 'remove-icon');
+    removeButton.setAttribute('id', 'remove-icon');
     removeButton.innerHTML = '🗑';
```
Comment (divinecharlotte) [inline src/modules/print.js:119]: - [ ] kindly consider updating the local storage when a user deletes a todo with this icon so as to meet the below requirement
> The updated array of items is stored in local storage, so the user gets the correct list values after the page reloads. Make sure to reload the page to test the array is stored.

consider doing the same even on the check boxes status by making sure that if the user updates the list when the UI is changing it also updates the local storage's  completed status

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @Kaiserabbas    , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## To Highlight 💯 

- Linters are passing👍 
- Descriptive pr👍 

 




 

## Required Changes ♻️

_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/print.js (hunk 2 above)
Comment (Kaiserabbas) [reply src/modules/print.js:119]: Save task is added to the code section to update local storage.
[
image
](url)

File: src/modules/print.js (hunk 1 above)
Comment (Kaiserabbas) [reply src/modules/print.js:160]: Filter method is used for "clear all completed" button
[
image
](url)

**Comment by Kaiserabbas** [comment]
1.  Filter method is used for "clear all completed" button

[https://github.com/Kaiserabbas/todo-list/assets/123939543/2d44f53c-a5c0-4899-81ed-360f839dcae9](url)

2. Save task is added to the code section to update local storage.
[]( https://github.com/Kaiserabbas/todo-list/assets/123939543/cc4d65da-a55c-421e-95a3-f0d98fd6cae8
)

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @Kaiserabbas , 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Clean code
✅ ... Great design

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**